sql/drop.go

```diff
@@ -211,6 +211,21 @@ func (n *dropIndexNode) Start() error {
 		// Queue the mutation.
 		switch status {
 		case sqlbase.DescriptorActive:
+			idx := tableDesc.Indexes[i]
+
+			// If an index is either end of a FK constraint, it cannot be dropped.
```

Comment:
If `n.n.DropBehavior == parser.DropCascade`, then the FK should be dropped as well to allow the `DROP INDEX` to proceed. We don't need to implement this now, but we might want to use a different error message to acknowledge that this is not yet supported. (Some frameworks add `CASCADE` to all their `DROP` commands so we want to accept that whenever there is nothing to actually cascade to).